Gems/Multiplayer/Code/Source/Components/NetworkHierarchyRootComponent.cpp

```diff
@@ -326,4 +331,99 @@ namespace Multiplayer
             RebuildHierarchy();
         }
     }
+
+    NetworkHierarchyRootComponentController::NetworkHierarchyRootComponentController(NetworkHierarchyRootComponent& parent)
+        : NetworkHierarchyRootComponentControllerBase(parent)
+    {
+
+    }
+
+    void NetworkHierarchyRootComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+
+    }
+
+    void NetworkHierarchyRootComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+
+    }
+
+    Multiplayer::MultiplayerController::InputPriorityOrder NetworkHierarchyRootComponentController::GetInputOrder() const
+    {
+        return Multiplayer::MultiplayerController::InputPriorityOrder::SubEntities;
+    }
+
+    void NetworkHierarchyRootComponentController::CreateInput(Multiplayer::NetworkInput& input, float deltaTime)
+    {
+        NetworkHierarchyRootComponent& component = GetParent();
+        if(!component.IsHierarchicalRoot())
+        {
+            return;
+        }
+
+        const AZStd::vector<AZ::Entity*>& entities = component.GetHierarchicalEntitiesRef();
```

Comment:
Ah, I see, you need them here. Is controller not a friend of the component base? We can definitely add a private method between them to better perf.